src/kv/MemDB.cc

```diff
@@ -425,6 +425,10 @@ bool MemDB::MDBWholeSpaceIteratorImpl::raw_key_is_prefixed(
     const string &prefix)
 {
   string p, k;
+  size_t pos = m_key_value.first.find(KEY_DELIM, 0);
+  if (pos == std::string::npos) {
+    return false;
+  }
```

Comment:
After checking for iterator invalidation, I no longer hit the 'assert_fail'. So I will delete this commit.